packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -99,76 +89,47 @@ export class EVMAAWallet extends LoggerWrapper {
-                        const { request } = await publicClient.simulateContract({
-                            account: this.account,
-                            address: contractAddress,
-                            abi: erc721,
-                            functionName: "safeTransferFrom",
-                            args: [this.getAddress(), toAddress, tokenId],
-                            ...(usesGelatoBundler(this.chain) && gelatoBundlerProperties),
-                        });
-                        transaction = await publicClient.writeContract(request);
-                        break;
-                    }
-                    default: {
-                        throw new WalletSdkError(`Token not supported`);
-                    }
-                }
+            if (!isAddress(toAddress)) {
+                throw new Error(`Invalid recipient address: '${toAddress}' is not a valid EVM address.`);
+            }
 
-                if (transaction != null) {
-                    return transaction;
-                } else {
-                    throw new TransferError(
-                        `Error transferring token ${evmToken.contractAddress}
-                    ${!tokenId ? "" : ` tokenId=${tokenId}`}
-                    ${!transaction ? "" : ` with transaction hash ${transaction}`}`
-                    );
-                }
+            if (!isAddress(config.token.contractAddress)) {
+                throw new Error(
+                    `Invalid contract address: '${config.token.contractAddress}' is not a valid EVM address.`
+                );
+            }
+
+            const tx = {
+                ...transferParams({
+                    contract: config.token.contractAddress,
+                    to: toAddress,
+                    from: this.account,
+                    config,
+                }),
+            };
+
+            try {
+                const client = this.smartAccountClient.extend(publicActions);
+                const { request } = await client.simulateContract(tx);
+                return client.writeContract(request);
             } catch (error) {
                 logError("[TRANSFER] - ERROR_TRANSFERRING_TOKEN", {
                     service: SCW_SERVICE,
                     error: errorToJSON(error),
-                    tokenId: tokenId,
-                    contractAddress: evmToken.contractAddress,
-                    chain: evmToken.chain,
+                    tokenId: tx.tokenId,
+                    contractAddress: config.token.contractAddress,
+                    chain: config.token.chain,
                 });
                 throw new TransferError(
-                    `Error transferring token ${evmToken.contractAddress}${tokenId == null ? "" : `:${tokenId}}`}`
+                    `Error transferring token ${config.token.contractAddress}${
+                        tx.tokenId == null ? "" : `:${tx.tokenId}}`
+                    }`
```

Comment:
super super nit: I prefer to add variables for this so that the string isn't so broken up & harder to read
```suggestion
const tokenId = tx.tokenId == null ? "" : `:${tx.tokenId}}`;
throw new TransferError(`Error transferring token ${config.token.contractAddress}${tokenId}`);
```

Reply:
agree that's better, was like that before but I fixed!